### PR TITLE
Use @nextcloud/webpack-vue-config and update other templating files

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: webpack.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const webpackConfig = require('@nextcloud/webpack-vue-config')
+
+const config = {
+	entry: {
+		admin: path.join(__dirname, 'src', 'admin.js'),
+		user: path.join(__dirname, 'src', 'user.js'),
+		public: path.join(__dirname, 'src', 'public.js'),
+	},
+	output: {
+		filename: 'terms_of_service_[name].js',
```
Comment (skjnldsv) [inline webpack.js:12]: Why?

Comment (skjnldsv) [reply webpack.js:12]: Using the default should be fine, no?

File: webpack.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { merge } = require('webpack-merge')
+const path = require('path')
+const webpackConfig = require('@nextcloud/webpack-vue-config')
+
+const config = {
+	entry: {
+		admin: path.join(__dirname, 'src', 'admin.js'),
+		user: path.join(__dirname, 'src', 'user.js'),
+		public: path.join(__dirname, 'src', 'public.js'),
```
Comment (skjnldsv) [inline webpack.js:9]: Don't forget to update the controllers to inject the proper script name as it changed

Comment (skjnldsv) [reply webpack.js:9]: AH no, strange
https://github.com/nextcloud/terms_of_service/blob/a0180066d6c19cf78bb5999ef878020ed3f52da9/lib/AppInfo/Application.php#L80
https://github.com/nextcloud/terms_of_service/blob/a0180066d6c19cf78bb5999ef878020ed3f52da9/lib/AppInfo/Application.php#L83

Where is the admin one used though?

Comment (skjnldsv) [reply webpack.js:9]: ok found https://github.com/nextcloud/terms_of_service/blob/6cffa5954ed1089aa161bacb29ffe81d40c08dd6/templates/settings.php#L22

File: webpack.js (hunk 1 above)
Comment (nickvergessen) [reply webpack.js:12]: So I dont need to change php code :see_no_evil: and the default would be `terms_of_service-[name].js`

File: webpack.js (hunk 2 above)
Comment (nickvergessen) [reply webpack.js:9]: it didn't thats why I overwrote it

Comment (skjnldsv) [reply webpack.js:9]: Yep I finally understood why you did it :p

**Pull request merged**